src/components/LoginHandler/LoginHandler.test.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { shallow, mount } from "enzyme";
+import LoginHandler from "./LoginHandler";
+import AuthenticationInfo from "AuthenticationInfo";
+import util from "util";
+
+let loginHandler;
+let loginHandlerInstance;
+const mockUpdateAuthenticationInfo = jest.fn();
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+it("renders LoginHandler without crashing", () => {
+  const wrapper = mount(<LoginHandler />);
+
+  expect(wrapper).toMatchSnapshot();
+
+  wrapper.unmount();
+});
+
+it("expects updateAuthenticationInfo to be called when LoginHandler is mounted", () => {
+  const lwaResponse = "access_token=some_access_token&expires_in=30";
+  const routeProps = { location: { hash: lwaResponse } };
+  const wrapper = mount(
+    <LoginHandler
+      updateAuthenticationInfo={mockUpdateAuthenticationInfo}
+      {...routeProps}
+    />
+  );
+
+  expect(mockUpdateAuthenticationInfo).toHaveBeenCalledTimes(1);
+  expect(
+    mockUpdateAuthenticationInfo.mock.calls[0][0].getAccessToken()
+  ).toEqual("some_access_token");
+
+  wrapper.unmount();
+});
+
+it("expects updateAuthenticationInfo to not be called when LoginHandler is mounted", () => {
+  const lwaResponse = "error_token=some_access_token&expires_in=30";
```

Comment:
Can the mounting itself be done in a loop by passing a different routeProps object each time (as against mounting them separately five times)?

Reply:
Updated the tests